code/jl/finite_lq.jl

```diff
@@ -69,47 +69,44 @@ function value_iteration(model; tol=1e-5)
     return get_greedy(v_star, model)
 end
 
-
-
 "Get the value v_σ of policy σ."
 function get_value(σ, model)
     # Unpack and set up
     (; β, a_0, a_1, γ, c, y_grid, z_grid, Q) = model
-    yn, zn = length(y_grid), length(z_grid)
-    n = yn * zn
-    # Function to extract (i, j) from m = i + (j-1)*yn"
-    single_to_multi(m) = (m-1)%yn + 1, div(m-1, yn) + 1
-    # Allocate and create single index versions of P_σ and r_σ
-    P_σ = zeros(n, n)
-    r_σ = zeros(n)
-    for m in 1:n
-        i, j = single_to_multi(m)
-        y, z, y′ = y_grid[i], z_grid[j], y_grid[σ[i, j]]
-        r_σ[m] = (a_0 - a_1 * y + z - c) * y - γ * (y′ - y)^2
-        for m′ in 1:n
-            i′, j′ = single_to_multi(m′)
-            if i′ == σ[i, j]
-                P_σ[m, m′] = Q[j, j′]
+    n_y, n_z = length(y_grid), length(z_grid)
```

Comment:
@jstac is `book-dp1` or `book-dp1-public` the source of truth for `finite_lq`. Do you recal?

Reply:
Looks like this code is the latest -- and we should port this back down to `book-dp1`